jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/skaffold/FilesMojoV2.java

```diff
@@ -0,0 +1,207 @@
+        // done if <packaging>pom</packaging>
+        continue;
+      }
+
+      // Add sources directory (resolved by maven to be an absolute path)
+      skaffoldFilesOutput.addInput(Paths.get(project.getBuild().getSourceDirectory()));
+
+      // Add resources directory (resolved by maven to be an absolute path)
+      project
+          .getBuild()
+          .getResources()
+          .stream()
+          .map(FileSet::getDirectory)
+          .map(Paths::get)
+          .forEach(skaffoldFilesOutput::addInput);
+
+      // This seems weird, but we will only print out the jib "extraFiles" directory on projects
+      // where the plugin is explicitly configured (even though _skaffold-files-v2 is a
+      // jib-maven-plugin goal and is expected to run on all projects irrespective of their
+      // configuring of the jib plugin).
+      if (project.getPlugin(MavenProjectProperties.PLUGIN_KEY) != null) {
+        // Add extra directory
+        skaffoldFilesOutput.addInput(resolveExtraDirectory(project));
+      }
+
+      // Grab non-project SNAPSHOT dependencies for this project
+      // TODO: this whole sections relies on internal maven API, it could break. We need to explore
+      // TODO: better ways to resolve dependencies using the public maven API.
+      Set<String> projectArtifacts =
+          projects
+              .stream()
+              .map(MavenProject::getArtifact)
+              .map(Artifact::toString)
+              .collect(Collectors.toSet());
+
+      DependencyFilter ignoreProjectDependenciesFilter =
+          (node, parents) -> {
+            if (node == null || node.getDependency() == null) {
+              // if nothing, then ignore
+              return false;
+            }
+            if (projectArtifacts.contains(node.getArtifact().toString())) {
+              // ignore project dependency artifacts
+              return false;
+            }
+            // we only want compile/runtime deps
+            return Artifact.SCOPE_COMPILE_PLUS_RUNTIME.contains(node.getDependency().getScope());
+          };
+
+      try {
+        DependencyResolutionResult resolutionResult =
+            projectDependenciesResolver.resolve(
+                new DefaultDependencyResolutionRequest(project, session.getRepositorySession())
+                    .setResolutionFilter(ignoreProjectDependenciesFilter));
+        resolutionResult
+            .getDependencies()
+            .stream()
+            .map(org.eclipse.aether.graph.Dependency::getArtifact)
+            .filter(org.eclipse.aether.artifact.Artifact::isSnapshot)
+            .map(org.eclipse.aether.artifact.Artifact::getFile)
+            .map(File::toPath)
+            .forEach(skaffoldFilesOutput::addInput);
+
+      } catch (DependencyResolutionException ex) {
+        throw new MojoExecutionException("Failed to resolve dependencies", ex);
+      }
+    }
+
+    try {
+      // Print JSON string
+      System.out.println("BEGIN JIB JSON");
```

Comment:
should we do `"\nBEGIN JIB JSON"` incase the previous line didn't inject a newline?